src/components/carousel/index.jsx

```diff
@@ -90,20 +101,38 @@ const Carousel = ({
 		Children.map(children, (child) => (child?.type === subcomponentType ? child : null))
 	);
 
+	useEffect(() => {
+		setSlidesToShowInView(getSlidesToShowFromDom(id));
+		setSlide(getSlidesToShowFromDom(id));
```

Comment:
I'm just noticing this... considering that the user can pass the id, we should be using `useRef` to get a handle for the element in question otherwise `getElementById()` will possibly grab the wrong one page.  This would also apply to line ~151 even though it's not part of this issue.

Reply:
Id's must be unique, but I can update to use Refs

Reply:
gotcha. still working on this right? otherwise looks good to merge to me. IDs have to be unique so I don't see a massive risk here. but, from a react perspective, it makes sense

Reply:
IDs _should_ be unique, but there's nothing stopping someone from putting the same ID on something else on the page.  In that case, `getElementById()` returns the first item.

Reply:
https://dom.spec.whatwg.org/#dom-nonelementparentnode-getelementbyid

Reply:
gotcha. I feel like getElementById should just error out if it gets more than 1 heh